pkgs/development/libraries/qt-5/5.7/qtinstaller.nix

```diff
@@ -0,0 +1,34 @@
+{ qtSubmodule, fetchurl, qtdeclarative , qttools, mirror }:
+
+qtSubmodule {
+  name = "qtinstaller";
+  qtInputs = [ qtdeclarative qttools ];
+  version = "2.0.3";
+  src = fetchurl {
+      url = "${mirror}/official_releases/qt-installer-framework/2.0.3/qt-installer-framework-opensource-2.0.3-src.tar.gz";
+      sha256 = "003gwjg02isw8qjyka377g1ahlisfyi44l6xfa4hvvdgqqq0hy2f";
+      name = "qt-installer-framework-opensource-src-2.0.3.tar.gz";
+  };
+
+  outputs = [ "out" "dev" "doc" ];
+
+  installPhase = ''
```

Comment:
When you override a phase like this, the first line must be `runHook preInstall` and the last line must be `runHook postInstall`.